### PR TITLE
feat(install): the prompt library is opt-in on a bare install, with --prompt-library and --no-prompt-library

### PR DESCRIPTION
**Pull request by Fmarzochi**
## What Changed

- A bare `egc install` now treats the prompt library as opt-in. The question defaults to no (`[y/N]`), `egc install --prompt-library` adds the library to every detected tool without asking, `egc install --no-prompt-library` skips the question, and a headless run (CI, piped stdin) skips it with a note that names the flag. The bash and PowerShell installers mirror each other.
- `scripts/install-apply.js` parses both flags, forwards them to the shell installer on the bare path (`legacyInstallerArgs`), and refuses them next to an explicit `--profile`, `--modules`, `--with`, `--without`, `--config` or language selection, where the choice is already made.
- `egc install --target <tool> --profile full` keeps working per tool, and an install that already carries a profile keeps it on `egc auto-update` (the profile is read from the install-state, untouched here).
- Installation guide (bare install stage, installer steps, full profile stage, command reference), CHANGELOG and ROADMAP describe the new default.

## Why This Change

Engine first. A first install should deliver the runtime (memory, Session Mesh, Guardian, Token Crusher, MCP registration) and nothing the person did not ask for; the library of agents, skills, commands and rules is valuable but is a deliberate choice, not a default. Until now an interactive install answered yes to the library on a bare Enter, so most people got the whole catalog without deciding.

## Testing Done

- [x] Manual testing completed: `bash -n` on `scripts/install.sh`; `install-apply.js --prompt-library --target cursor --profile core --dry-run` refuses with the explanation; the bare path forwards exactly one flag or none.
- [x] Automated tests pass locally (`node tests/run-all.js`): 151 suites, 4838 tests, all passing (the two guardian suites needed the local server build refreshed first; the build was older than its sources)
- [x] Edge cases considered and tested: empty answer means no in both installers; a null `Read-Host` on a redirected stdin can no longer count as yes; the flags are never read as languages; a selection plus the flag fails before anything is planned.
- [x] Not applicable: no file shared across concurrent EGC processes is touched.

Targeted suites: install-request 8/8, install-onboarding 13/13, install-ps1 13/13, install-sh 9/9, selective-install 41/41, ci/catalog 7/7, egc 20/20. The PowerShell runtime path is covered by the Windows lanes in CI (no `pwsh` on the machine that ran the suites).

## Type of Change
- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [x] `docs:` Documentation
- [x] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly (none changed)
- [x] Shell scripts pass `bash -n`; shellcheck not available on this machine
- [x] Pre-commit hooks pass locally
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation
- [x] Added comments for complex logic
- [ ] README updated: follows in the repositioning PR, which describes the engine-first install and the optional library together

---
## Summary by cubic
Makes the prompt library opt-in on a bare `egc install`: the interactive question now defaults to no (`[y/N]`), `--prompt-library` installs to every detected tool without asking, `--no-prompt-library` skips the question, and a headless run skips it with a note naming the flag.

- `install-apply.js` forwards the flag to the shell installer on the bare path and refuses the flags next to an explicit `--profile`, `--modules`, `--with`, `--without`, `--config`, or language selection, and refuses the two flags together.
- The PowerShell installer now stops when a delegated targeted install fails, and the bash prompt is skipped under CI even with a terminal on stdin.
- `egc install --target <tool> --profile full` keeps working per tool, and an existing profile is kept on `auto-update`.
- Installation docs, `CHANGELOG`, and `ROADMAP` updated.

<sup>Written for commit 4f481ca8487f017c79bd8fcfb19d86636730962c. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Fmarzochi/EGC/pull/1440?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by Fmarzochi** [comment]
Second round pushed, one commit, all five findings taken:

- The pair `--prompt-library --no-prompt-library` is refused at parse time in either order (`applyPromptLibrary` in request.js), and both shell installers refuse it as well when run directly, so Windows and Unix behave the same and argument order never decides.
- The PowerShell installer exits on a non-zero delegated install, the way the bash installer stops under `set -e`.
- The bash prompt is gated on `-t 0 && -z "${CI:-}"`, matching the PowerShell gate.
- The installation guide says that `--prompt-library` installs a full profile per tool and therefore creates the install-state files the doctor and the repair track, and that the two flags cannot be combined.

Tests: the parser refuses the pair in both orders; the installer suite asserts the CI guard, the shared refusal text and the delegated exit; targeted suites green locally and the full suite rerun before the push.

File: scripts/install-apply.js [hunk 1]
```diff
@@ -191,10 +196,19 @@ function configureCommitPrivacyFilterBestEffort(log, onError) {
 // flow is already defined by the shipped onboarding installers, so run
 // them instead of failing; the env guard stops the wrappers from
 // recursing back into this script.
-function delegateToLegacyInstaller() {
+// The bare-install flags the shell installers understand. Kept apart from
+// the spawn so the argument shape has a test of its own.
+function legacyInstallerArgs(options = {}) {
+  if (options.promptLibrary === true) return ['--prompt-library'];
```
[the remark this answers is not in the document]
Comment (Fmarzochi) [reply scripts/install-apply.js:202]: Addressed in 4f481ca8: the parser refuses the pair in either order (applyPromptLibrary), and both shell installers refuse it as well when run directly, so argument order never decides.
